ecl/eclscheduler/eclscheduler.cpp

```diff
@@ -180,6 +159,198 @@ version: "1.0"
   name: myeclscheduler
 )!!";
 
+//=========================================================================================
+
+class EclSchedulerServer : implements CSimpleInterfaceOf<IAbortHandler>
+{
+public:
+    EclSchedulerServer()
+    {
+        addAbortHandler(*this);
+    }
+    ~EclSchedulerServer()
+    {
+        if (updateConfigCBId)
+            removeConfigUpdateHook(updateConfigCBId);
+
+        removeAbortHandler(*this);
+    }
+
+    // IAbortHandler
+    virtual bool onAbort() override
+    {
+        DBGLOG("onAbort()");
+        running = false;
+        configChangeOrAbort.signal();
+
+#ifdef _DEBUG
+        return false; // we want full leak checking info
+#else
+        return true; // we don't care - just exit as fast as we can
+#endif
+    }
+
+    void run(Owned<IFile> &sentinelFile)
```

Comment:
simpler and more conventional/flexible to pass in the underlying type, i.e. IFile